tests/utils.py

```diff
@@ -13,29 +14,41 @@
 
 def recurse_call(func):
     def wrapper(*args, **kwargs):
+        last_error = None
         for _ in range(60):
             try:
                 result = func(*args, **kwargs)
                 break
-            except (HTTPStatusError, ValueError):
+            except (
+                HTTPStatusError,
+                ValueError,
+                ObjectNotFound,  # Lagging permissions
+            ) as e:
+                last_error = e
                 sleep(0.5)
         else:
-            raise TimeoutError
+            raise TimeoutError from last_error
         return result
 
     return wrapper
 
 
 def async_recurse_call(func):
     async def wrapper(*args, **kwargs):
+        last_error = None
         for _ in range(60):
             try:
                 result = await func(*args, **kwargs)
                 break
-            except (HTTPStatusError, ValueError):
+            except (
+                HTTPStatusError,
+                ValueError,
+                ObjectNotFound,  # Lagging permissions
```

Comment:
```suggestion
                ObjectNotFound,  # Lacking permissions
```

Reply:
The lagging was intentional: as in, the permissions are not yet set but will soon be. It's to address a race condition that sometimes happens on my (slow) system.

Reply:
Ah okay, then I didn't understand the comment at all. Might be better to put that explanation instead.

Reply:
Might be my gaming youth shining through here. Always complain about the lag ;-). I've improved the comments.